src/shared.zig

```diff
@@ -174,6 +174,59 @@ pub const MappedFile = struct {
     }
 };
 
+pub fn readFileIntoBuffer(
+    comptime subcommand: type,
+    allocator: std.mem.Allocator,
+    io: anytype,
+    cwd: std.fs.Dir,
+    path: []const u8,
+    buffer: []u8,
+) error{ AlreadyHandled, OutOfMemory }![]const u8 {
+    const file = cwd.openFile(path, .{}) catch {
+        return printErrorAlloc(
+            subcommand,
+            allocator,
+            io,
+            "unable to open '{s}'",
+            .{path},
+        );
+    };
+    defer if (free_on_close) file.close();
+
+    const stat = file.stat() catch |err| {
+        return printErrorAlloc(
+            subcommand,
+            allocator,
+            io,
+            "unable to stat '{s}': {s}",
+            .{ path, @errorName(err) },
+        );
+    };
+
+    if (stat.size == 0) {
+        return printErrorAlloc(
+            subcommand,
+            allocator,
+            io,
+            "file is empty '{s}'",
+            .{path},
+        );
+    }
```

Comment:
I don't think this helper needs to stat the file, we should instead just allow the `readAll` call to return zero resulting in an zero length slice being returned.

For nprocs it will mean `if (it.next() == null) return error.InvalidFormat;` in `getLastCpuIndex` will be hit which is fine, or maybe a check for non-zero length after the `readFileIntoBuffer` call.

Reply:
Since it is already caught by the split and because `file_contents` is used nowhere else in `execute`, I think it's enough to add a comment.